client/ayon_core/hosts/hiero/api/workio.py

```diff
@@ -51,12 +51,11 @@ def open_file(filepath):
 
     project = hiero.core.projects()[-1]
 
-    # open project file
-    hiero.core.openProject(filepath.replace(os.path.sep, "/"))
-
-    # close previous project
-    project.close()
-
+    # Close previous project if its different to the current project.
+    if project.path() != filepath:
+        # open project file
+        hiero.core.openProject(filepath.replace(os.path.sep, "/"))
```

Comment:
```suggestion
    # Close previous project if its different to the current project.
    filepath = filepath.replace(os.path.sep, "/")
    if project.path() != filepath:
        # open project file
        hiero.core.openProject(filepath)
```
I tested with multiple scenario and i find out that the code still errors out when I have unsaved changes and click the same workfile and click `No`. And the suggestion can get it fix for that scenario.